packages/aws-library/src/aws_library/s3/_error_handler.py

```diff
@@ -13,6 +13,11 @@
     S3UploadNotFoundError,
 )
 
+if TYPE_CHECKING:
+    # NOTE: TYPE_CHECKING is True when static type checkers are running,
+    # allowing for circular imports only for them (mypy, pylance, ruff)
+    from ._client import SimcoreS3API
```

Comment:
So this is a way to "forward declare" `SimcoreS3API` just for the purpose of type-checking, right?

Reply:
yes